index/scorch/scorch.go

```diff
@@ -139,17 +139,18 @@ func NewScorch(storeName string,
 // configForceSegmentTypeVersion checks if the caller has requested a
 // specific segment type/version
 func configForceSegmentTypeVersion(config map[string]interface{}) (string, uint32, error) {
-	forcedSegmentVersion, ok := config["forceSegmentVersion"].(int)
-	if ok {
-		forcedSegmentType, ok2 := config["forceSegmentType"].(string)
-		if !ok2 {
-			return "", 0, fmt.Errorf(
-				"forceSegmentVersion set to %d, must also specify forceSegmentType", forcedSegmentVersion)
-		}
+	forcedSegmentVersion, err := parseToInteger(config["forceSegmentVersion"])
+	if err != nil {
+		return "", 0, nil
```

Comment:
in case of err, its returning nil?

Reply:
Yes, because error in this case just means that it was not int or float64, in which case we do not proceed to process the settings (which I believe is consistent with what we did before).

Reply:
I am not very sure of this.  Giving some hints about the incorrect format looks cleaner to me than not giving any clues to the user of what is going wrong.
Also, unable to comprehend the importance of maintaining consistency with what we did before for a small period of time between v11 - v12/13..

Reply:
I'm not arguing for consistency, I'm simply saying I'm not changing the behavior (other than adding support for float64).  If you'd like to improve the error handling, maybe you should create your own PR?